scripts/extensions/datetimeField/src/getToggleTemplateDateTimeField.tsx

```diff
@@ -8,34 +8,45 @@ interface IPropsAdditional {
 }
 
 export function getToggleDateTimeField(superdesk: ISuperdesk) {
-    const {notify} = superdesk.ui;
     const {gettext} = superdesk.localization;
 
     return class ToggleDateTimeField
         extends React.PureComponent<IEditorComponentProps<string | null, IDateTimeFieldConfig> & IPropsAdditional> {
```

Comment:
Type type should be `React.PureComponent<ITemplateEditorComponentProps<...`, not `React.PureComponent<IEditorComponentProps<...`. I see `ITemplateEditorComponentProps` got deleted, so it should be restored. Also remove `template` prop from `ITemplateEditorComponentProps` - editor should already receive a value with template applied if needed.

Reply:
> 

We have removed getTemplateDateTimeField.tsx file. I think we can use `IEditorComponentProps` because `ITemplateEditorComponentProps` and `IEditorComponentProps` both are seems to be same interface so any specific reason to use I`TemplateEditorComponentProps` here?

Reply:
It seems to me that `getTemplateDateTimeField` was replaced with `getToggleTemplateDateTimeField.tsx`, but the usage remains the same - it's a component that is only used for managing the template and is not used in the editor, right?

> any specific reason to use ITemplateEditorComponentProps here?

To reduce code coupling. It's cleaner from code quality point of view, to contain features so they don't know about each other and can't access each other's code or data. Isolating functionality in one place, or at least in as little places as possible makes code more stable and easier to reason about. Depending on requirements it is sometimes not possible, but we should isolate them when possible as it is now.